src/store/authors/thunk.js

```diff
@@ -0,0 +1,33 @@
+import APIService from '../../util/APIService';
+import { ENDPOINTS } from '../../util/consts';
+import { addAuthor, addAuthors, deleteAuthor } from './actionCreators';
+
+export const getAllAuthors = () => (dispatch) =>
+	APIService.Get(ENDPOINTS.GET_ALL_AUTHORS)
+		.then((authors) => dispatch(addAuthors(authors)))
+		.catch((err) => {
+			// eslint-disable-next-line no-alert
+			alert('you are not logged in as admin');
+		});
+
+export const postAuthor = (authData, token) => (dispatch) => {
```

Comment:
You can get token inside your thunk function from getState (second parametr after dispatch). 
`postAuthor = (authData, token) => (dispatch) => {`